sklearn/utils/sparsefuncs_fast.pyx

```diff
@@ -309,13 +312,16 @@ def inplace_csr_row_normalize_l1(X):
 @cython.wraparound(False)
 @cython.cdivision(True)
 def inplace_csr_row_normalize_l2(X):
-    """Inplace row normalize using the l2 norm"""
-    cdef unsigned int n_samples = X.shape[0]
-    cdef unsigned int n_features = X.shape[1]
+    _inplace_csr_row_normalize_l2(X.data, X.shape, X.indices, X.indptr)
```

Comment:
Why I put a wrapper here is because `inplace_csr_row_normalize_l1` and `inplace_csr_row_normalize_l2` only accepts one argument `X`, it is not allowed to directly replace

 ```python
cdef np.ndarray[DOUBLE, ndim=1] X_data = X.data
``` 

with 

```python
cdef np.ndarray[floating, ndim=1] X_data = X.data
``` 
However, it is available in case like #6430 where function also accepts some arguments that has type involves `floating`. e.g. `np.ndarray[floating, ndim=2]`